corehq/apps/export/models/new.py

```diff
@@ -712,6 +712,15 @@ class ExportInstance(BlobMixin, Document):
     class Meta(object):
         app_label = 'export'
 
+    @classmethod
+    def wrap(cls, data):
+        from corehq.apps.export.views.utils import clean_odata_columns, remove_row_number_from_export_columns
+        export_instance = super(ExportInstance, cls).wrap(data)
+        if export_instance.is_odata_config:
+            clean_odata_columns(export_instance)
+            remove_row_number_from_export_columns(export_instance)
+        return export_instance
```

Comment:
This feels a little funny to me to be happening inside the wrap method. Is there not a place closer to where it is accessed that we could do the transform? Also, does mutating `.tables` directly not change the underlying object? Like I would have thought that the following:

```python
export = ExportInstance.get(some_odata_object)
export.save()
```

Might actually change/break the underlying JSON since you are modifying the `.tables` in place. Is that wrong?

Seems like a clearer/safer workflow would be something more like:

```python
export = ExportInstance.get(some_odata_object)
transformed_data = transform_for_odata(export)
# do whatever you need with transformed data
```

Reply:
@czue I added tests that document that the overridden ```wrap``` functions are working: 26a2c9c

The reason I went with overriding ```wrap``` is that I want to ensure that all OData feeds fetched from the database are properly cleaned. Otherwise it seems too easy for feature dev work to miss this.

Reply:
I don't think the tests you added address the concern I had. I just added a failing test that does here: https://github.com/dimagi/commcare-hq/pull/24847/commits/adb8ea10ce8a64b94073de83f7f3b3eb9e03578e

Is it by design that calling `.get` followed by `.save` would modify the underlying doc? What implications does that have in the UI? It still seems very unexpected and like not ideal behavior to me.